recipes/openssl/1.x.x/conanfile.py

```diff
@@ -683,16 +683,20 @@ def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "OpenSSL"
         self.cpp_info.names["cmake_find_package_multi"] = "OpenSSL"
         if self._use_nmake:
+            libsuffix = "d" if self.settings.build_type == "Debug" else ""
             if self._full_version < "1.1.0":
-                self.cpp_info.libs = ["ssleay32", "libeay32"]
+                self.cpp_info.components["SSL"].libs = ["ssleay32" + libsuffix]
+                self.cpp_info.components["Crypto"].libs = ["libeay32" + libsuffix]
             else:
-                if self.settings.build_type == "Debug":
-                    self.cpp_info.libs = ['libssld', 'libcryptod']
-                else:
-                    self.cpp_info.libs = ['libssl', 'libcrypto']
+                self.cpp_info.components["SSL"].libs = ["libssl" + libsuffix]
+                self.cpp_info.components["Crypto"].libs = ["libcrypto" + libsuffix]
         else:
-            self.cpp_info.libs = ["ssl", "crypto"]
+            self.cpp_info.components["SSL"].libs = ["ssl"]
+            self.cpp_info.components["Crypto"].libs = ["crypto"]
+ 
         if self.settings.os == "Windows":
-            self.cpp_info.system_libs.extend(["crypt32", "msi", "ws2_32", "advapi32", "user32", "gdi32"])
+            self.cpp_info.components["SSL"].system_libs.extend(["crypt32", "ws2_32"])
```

Comment:
Is there any place to verify the system libs?

Reply:
Filling system libs automatically ? It would be nice, but I am not sure it would be possible.

Reply:
I have something here for verification: https://github.com/conan-io/hooks/pull/171
But its tests are failing on Appveyor because it cannot find the Visual Studio toolchain (which it needs to analyse the binaries)

Reply:
@madebr Really nice !

Reply:
The pr is in limbo because I don't know how to let the tests pass on Windows.
So if any of you have an idea how to fix this, it would be very appreciated!